3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +22,69 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(`Response failed!`);
+    }
+    return response.json();
+  } catch (error) {
+    console.log(error.message);
+    throw error;
+  }
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchAndPopulatePokemons() {
+  try {
+    const url = 'https://pokeapi.co/api/v2/pokemon?offset=151&limit=151';
+    const data = await fetchData(url);
+    const select = document.createElement('select');
+    select.classList.add('select-pokemon');
+    for (const result of data.results) {
+      const option = document.createElement('option');
+      option.textContent = result.name;
+      option.value = result.url;
+      select.appendChild(option);
+    }
+    document.body.appendChild(select);
+    select.addEventListener('change', () => {
+      fetchImage(select.value);
+    });
+  } catch (error) {
+    console.log(error);
+  }
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchImage(url) {
+  try {
+    const data = await fetchData(url);
+    if (document.querySelector('img')) {
+      document.querySelector('body')
+        .removeChild(document.querySelector('img'));
+    }
+    const image = document.createElement('img');
```

Comment:
This alternative would work too. Instead of removing an existing `<img>` you could just reuse it.

```suggestion
    let image =  document.querySelector('img');
    if (!image) {
       image = document.createElement('img');
       document.body.appendChild(image);
    }
```